app/controllers/api/v1/posts_controller.rb

```diff
@@ -0,0 +1,11 @@
+class Api::V1::PostsController < ApplicationController
+  def index
+    if params[:user_id]
+      user = User.find(params[:user_id])
+      posts = user.posts
+    else
+      posts = Post.all
+    end
+    render json: posts
+  end
```

Comment:
- You did well so far, but I have some errors when trying to create a post, this issue comes from your Post, according to your schema file, your post should belong to an `author` not user. Please fix this 🙏. Also remember to specify the class_name of the author 👍 👍 
![eathsrh](https://github.com/Rifat8080/blog-app/assets/73607512/48856373-7e57-4cd9-aaed-531f9109fedd)
![rhaerghsers](https://github.com/Rifat8080/blog-app/assets/73607512/9415cbbe-0621-45ad-b34d-2d450b48aba3)


- Please correct the `update_user_counter` method to use `author` instead of `user` after making the required change above 👍 👍